### DISCUSSION TITLE
Best practice for isLoading. What different with isValidating?

### DISCUSSION THREAD

**Discussion by terrierscript** [vercel/swr#563] [General] [answered]
In SWR documentation, I cannot found about `isLoading` explicitly.

[This page]( https://swr.vercel.app/getting-started#make-it-reusable) show like this

```
isLoading: !error && !data,
```

Similirary, [quick start] section show like this
https://swr.vercel.app/getting-started#quick-start
```js
 if (error) return <div>failed to load</div>
 if (!data) return <div>loading...</div>
```

And [`isValidating`](https://swr.vercel.app/docs/options#return-values) is explained like this

> `isValidating: if there's a request or revalidation loading`

Question1:  What is best practice for isLoading? Is this `isLoading: !error && !data` ?
Question2: What is `isValidating` use case?

**Comment by sergiodxa** [comment, marked as answer]
Loading only happens once, when you don’t have data or error, this is when it’s the first time you use a key.

Validation can happen more than once, when you already have data or error and SWR is calling your fetcher again to update it.

I usually use Suspense for loading, but you can check if if you have data or error as in the examples (I like more the second one checking for error first and !data later).

You should use isValidating if you want to tell your users your data is being updated.
